apps/backend/src/applications/applications.service.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  BadRequestException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { MongoRepository } from 'typeorm';
+import { UserStatus } from '../users/types';
+import { UsersService } from '../users/users.service';
+import { getCurrentUser } from '../users/utils';
+import { Application } from './application.entity';
+import { getCurrentCycle } from './utils';
+import { Cycle } from './dto/cycle.dto';
+import { plainToClass } from 'class-transformer';
+
+@Injectable()
+export class ApplicationsService {
+  constructor(
+    @InjectRepository(Application)
+    private applicationsRepository: MongoRepository<Application>,
+    private readonly usersService: UsersService,
+  ) {}
+
+  async findOne(userId: number): Promise<Application> {
+    const currentUser = getCurrentUser();
+    const currentStatus = currentUser.status;
+    switch (currentStatus) {
+      case UserStatus.ADMIN:
+      case UserStatus.RECRUITER:
+        break;
+      default:
+        if (currentUser.userId !== userId) {
+          throw new UnauthorizedException('User not found');
+        }
+        break;
+    }
+
+    const applicant = await this.usersService.findOne(userId);
+    const application = applicant.applications[0];
```

Comment:
Hmm yeah we do something like this for SFTT and it's been fine so I don't think it's too big of an issue but any given user won't have more than a couple of apps (so doing more complex operations for this won't affect its runtime to a noticeable extent) and this is a common operation so I think this is a good idea! I did change the logic so that it actually gets the app for the current cycle instead of the latest app because we shouldn't need to look at apps for past cycles often. Maybe in a future iteration we can modify this endpoint so that it takes in a cycle year and semester so that we can also get an app for an arbitrary cycle